django/db/models/expressions.py

```diff
@@ -547,6 +550,53 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None,
         return ResolvedOuterRef(self.name)
 
 
+class SliceableF(F):
+    """
+    An object that can add slicing notation to F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f, k):
+        if not isinstance(k, (int, slice)):
+            raise TypeError
+        assert ((not isinstance(k, slice) and (k >= 0)) or
```

Comment:
`assert` cannot safely be used since `python -O` removes these statements. use `if`, and raise a `ValueError` since these are errors around the values passed in.

Also django code style doesn't use `\` line continuations, see https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/

Reply:
I have cleaned up this quite a bit, as well as moved to raising `ValueError` instead.